lib/client.js

```diff
@@ -31,10 +31,19 @@ function defaultId () {
 function sendPacket (client, packet, cb) {
   try {
     var buf = mqttPacket.generate(packet);
-    if (!client.stream.write(buf) && cb) {
-      client.stream.once('drain', cb);
-    } else if (cb) {
-      cb();
+
+    if (client.stream.write(buf)) {
+      client.emit('packetsent', packet.topic, packet.payload, packet);
+      if (cb) {
+        cb();
+      }
+    } else {
+      client.stream.once('drain', function () {
+        client.emit('packetsent', packet.topic, packet.payload, packet);
+        if (cb) {
+          cb();
+        }
+      });
```

Comment:
waiting "drain" does not really help in guaranteeing a packet was sent. It's used only for handling backpressure.
I would avoid allocating a function and emitting the event anyway, without waiting.

Also, I would just have `client.emit('packetsent', packet)`.